src/__swaps__/utils/flipAssets.ts

```diff
@@ -0,0 +1,137 @@
+import { SharedValue } from 'react-native-reanimated';
+import { inputKeys, inputMethods, inputValuesType } from '@/__swaps__/types/swap';
+import { valueBasedDecimalFormatter } from '@/__swaps__/utils/decimalFormatter';
+import { niceIncrementFormatter } from '@/__swaps__/utils/swaps';
+import { ExtendedAnimatedAssetWithColors } from '@/__swaps__/types/assets';
+import { divWorklet, equalWorklet, greaterThanWorklet, mulWorklet } from '@/__swaps__/safe-math/SafeMath';
+
+export function getInputValuesForSliderPositionWorklet({
+  selectedInputAsset,
+  percentageToSwap,
+  sliderXPosition,
+}: {
+  selectedInputAsset: ExtendedAnimatedAssetWithColors | null;
+  percentageToSwap: number;
+  sliderXPosition: number;
+}) {
+  'worklet';
+  const inputAssetMaxSwappableBalance = selectedInputAsset?.maxSwappableAmount || 0;
+  const isStablecoin = selectedInputAsset?.type === 'stablecoin';
+
+  const inputAmount = niceIncrementFormatter({
+    inputAssetBalance: inputAssetMaxSwappableBalance,
+    inputAssetNativePrice: selectedInputAsset?.price?.value ?? 0,
+    percentageToSwap,
+    sliderXPosition,
+    stripSeparators: true,
+    isStablecoin,
+  });
+
+  const inputNativeValue = mulWorklet(inputAmount, selectedInputAsset?.price?.value ?? 0);
+
+  return {
+    inputAmount,
+    inputNativeValue,
+  };
+}
+
+export const updateInputValuesAfterFlip = ({
```

Comment:
this is the main logic that is new / updated for how to handle input values after a flip